src/conditions.ts

```diff
@@ -214,13 +215,14 @@ export class ConditionMonitor extends EventEmitter {
     }
 
     try {
+      // console.log("condition", condition);  
       if (match) {
         await condition.onMatched(emittedValue);
-        await condition.sdkOnMatched();
+        // await condition.sdkOnMatched();
```

Comment:
This is commented because sdkOnMatched can be undefined correct? If so we can use an optional chained call so it does not break

```suggestion
        await condition.sdkOnMatched?.();
```